### PR TITLE
Migrate to standalone docs repo

### PR DESCRIPTION
**Pull request by charliermarsh**
## Summary

See: https://github.com/astral-sh/uv/pull/5081

### CODE REVIEW

File: .github/workflows/publish-docs.yml [hunk 1]
```diff
@@ -21,42 +21,119 @@ jobs:
...
-      - name: "Deploy to Cloudflare Pages"
-        if: ${{ env.CF_API_TOKEN_EXISTS == 'true' }}
-        uses: cloudflare/wrangler-action@v3.7.0
-        with:
-          apiToken: ${{ secrets.CF_API_TOKEN }}
-          accountId: ${{ secrets.CF_ACCOUNT_ID }}
-          # `github.head_ref` is only set during pull requests and for manual runs or tags we use `main` to deploy to production
-          command: pages deploy site --project-name=astral-docs --branch ${{ github.head_ref || 'main' }} --commit-hash ${GITHUB_SHA}
+
+      - name: "Clone repo"
```
Comment (MichaReiser) [inline .github/workflows/publish-docs.yml:94]:
```suggestion
      - name: "Clone docs repo"
```

**Review by MichaReiser — APPROVED** [review]

**Comment by MichaReiser** [comment]
Does this need any update to the contribution guidelines?

**Comment by charliermarsh** [comment]
Nope! No changes to anything at all. This only affects the publishing pipeline which happens during deploy.

**Comment by charliermarsh** [comment]
I won't be merging this until I change a few other things downstream though (and we feel ready to post the uv docs).

**Comment by dhruvmanila** [comment]
(Merging this as requested by Charlie.)

**Pull request merged**